scripts/odb_extractor_atoms.py

```diff
@@ -49,101 +223,311 @@ def select_obsclass(classes):
+
+    # Group by changes in exptimes/coadds?
+    # Write results to the Excel worksheet if given
+    if ws is not None:
+        # Columns
+        columns = ['datalab', 'class', 'type', 'inst', 'exec_time', 'exptime', 'coadds', 'fpu', 'filter',
+                   'disperser', 'wavelength', 'p', 'q', 'guiding', 'qa_state', 'atom']
+        row = 1
+        for jj, col in enumerate(columns):
+            _ = ws.cell(column=jj + 1, row=row, value="{0}".format(col))
+        row += 1
+
+    npattern = offset_lag
+    noffsets = 0
     for ii, step in enumerate(sequence):
         nextatom = False
-        qoffsets = False
 
         datalab = step['observe:dataLabel']
         if datalab in datalabels:
             qastate = obslog[datalab]['qaState']
         else:
             qastate = 'NONE'
-        qastates.append(qastate.upper())
 
         observe_class = step['observe:class']
-        classes.append(observe_class.upper())
 
-        exptime = float(step['observe:exposureTime'])
-        inst = step['instrument:instrument']
-        #     print(inst, fpuinst[inst])
-        fpu = step[fpuinst[inst]]
-        if 'instrument:filter' in step.keys():
-            filter = step["instrument:filter"]
-        else:
-            filter = 'None'
-        wavelength = float(step['instrument:observingWavelength'])
         step_time = step['totalTime'] / 1000.
 
-        if 'GMOS' in inst:
-            coadds = 1
-        else:
-            coadds = int(step['observe:coadds'])
-
-        disperser = step['instrument:disperser']
-        if 'telescope:p' in step.keys():
-            p = float(step['telescope:p'])
-        else:
-            p = 0.0
-        if 'telescope:q' in step.keys():
-            q = float(step['telescope:q'])
-            qoffsets = True
-        else:
-            q = 0.0
-
-            # Any wavelength/filter change is a new atom
-        if ii == 0 or (ii > 0 and wavelength != float(sequence[ii - 1]['instrument:observingWavelength'])):
+        atomstr = 'Atom for: '
+        # Any wavelength/filter change is a new atom
+        if ii == 0 or (ii > 0 and config['wavelength'][ii] != config['wavelength'][ii - 1]):
             nextatom = True
-            print('Atom for wavelength')
+            atomstr += 'wavelength, '
 
-        #         if observe_class == 'science':
-
-        # AB
-        # ABBA
-        if qoffsets and nsteps >= 4 and nsteps - ii > 3 and nabba == 0:
-            # print(q, sequence[ii+1]['telescope:q'], sequence[ii+2]['telescope:q'], sequence[ii+3]['telescope:q'])
-            if q == float(sequence[ii + 3]['telescope:q']) and q != float(sequence[ii + 1]['telescope:q']) and float(
-                    sequence[ii + 1]['telescope:q']) == float(sequence[ii + 2]['telescope:q']):
-                nabba = 3
+        # A change in exposure time or coadds is a new atom for science exposures
+        if step['observe:observeType'].upper() not in ['FLAT', 'ARC', 'DARK', 'BIAS']:
```

Comment:
```python
['FLAT', 'ARC', 'DARK', 'BIAS']
```
should be a predefined set. Otherwise it makes this list over and over and does list comparison instead of set comparison.